expression/builtin_other.go

```diff
@@ -106,11 +117,51 @@ func (c *inFunctionClass) getFunction(ctx sessionctx.Context, args []Expression)
 // builtinInIntSig see https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_in
 type builtinInIntSig struct {
 	baseBuiltinFunc
+	args      []Expression
+	hashSet   map[int64]bool
+	threshold int
+}
+
+func (b *builtinInIntSig) buildHashMapForConstArgs(ctx sessionctx.Context) error {
+	fmt.Println("enter buildHashMapForConstArgs")
+	b.args = make([]Expression, 0, len(b.baseBuiltinFunc.args))
+	b.args = append(b.args, b.baseBuiltinFunc.args[0])
+	b.hashSet = make(map[int64]bool, len(b.baseBuiltinFunc.args)-1)
+	count := 0
+	for i := 1; i < len(b.baseBuiltinFunc.args); i++ {
+		if b.baseBuiltinFunc.args[i].ConstItem() {
+			val, isNull, err := b.baseBuiltinFunc.args[i].EvalInt(ctx, chunk.Row{})
+			if err != nil {
+				return err
+			}
+			if isNull {
+				b.args = append(b.args, b.baseBuiltinFunc.args[i])
+				continue
+			}
+			b.hashSet[val] = mysql.HasUnsignedFlag(b.baseBuiltinFunc.args[i].GetType().Flag)
+			count++
+		} else {
+			b.args = append(b.args, b.baseBuiltinFunc.args[i])
+		}
+	}
+	fmt.Println("const count = ", count, ", threshold = ", b.threshold)
+	if count < b.threshold {
```

Comment:
You can write a benchmark to decide the threshold.

Reply:
I ran a benchmark and find the threshold is actually `1`
```
const count =  1 , threshold =  1
BenchmarkVectorizedBuiltinOtherFunc/builtinInIntSig-VecBuiltinFunc-12                     111224             10272 ns/op            1024 B/op          1 allocs/op
BenchmarkVectorizedBuiltinOtherFunc/builtinInIntSig-NonVecBuiltinFunc-12                   62086             18713 ns/op               0 B/op          0 allocs/op

const count =  1 , threshold =  200
BenchmarkVectorizedBuiltinOtherFunc/builtinInIntSig-VecBuiltinFunc-12                      52753             22356 ns/op            1024 B/op          1 allocs/op
BenchmarkVectorizedBuiltinOtherFunc/builtinInIntSig-NonVecBuiltinFunc-12                   16131             74576 ns/op               0 B/op          0 allocs/op
```
maybe it will be different in other types of `IN`